File: music-streaming-frontend/src/services/adminService.js
import api from './api';

export const adminService = {
  getAllUsers: async () => {
    const { data } = await api.get('/admin/users');
    return data;
  },
  getUsersByRole: async (role) => {
    const { data } = await api.get(`/admin/users/role/${role}`);
    return data;
  },
  getPendingArtists: async () => {
    const { data } = await api.get('/admin/artists/pending');
    return data;
  },
  approveArtist: async (artistId) => {
    const { data } = await api.post(`/admin/artists/${artistId}/approve`);
    return data;
  },
  rejectArtist: async (artistId) => {
    const { data } = await api.post(`/admin/artists/${artistId}/reject`);
    return data;
  },
  suspendUser: async (userId, reason) => {
    const { data } = await api.post(`/admin/users/${userId}/suspend`, reason);
    return data;
  },
  activateUser: async (userId) => {
    const { data } = await api.post(`/admin/users/${userId}/activate`);
    return data;
  },
  getUserStatistics: async () => {
    const { data } = await api.get('/admin/statistics/users');
    return data;
  }
};

File: deployment-backend/src/main/java/com/backend/service/AdminService.java
package com.backend.service;

import com.backend.dto.request.UserStatusUpdateRequest;
import com.backend.dto.response.SongResponse;
import com.backend.dto.response.UserResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;


public interface AdminService {
    
    // ========== User Management ==========
    
 
    UserResponse approveArtist(Long artistId);
    

    UserResponse rejectArtist(Long artistId);
    

    UserResponse suspendUser(Long userId, UserStatusUpdateRequest request);

    UserResponse activateUser(Long userId);
    
    List<UserResponse> getAllUsers();
    
    List<UserResponse> getUsersByRole(String role);
    
    List<UserResponse> getPendingArtists();
    
    Object getUserStatistics();
}
